builder_limit.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 The Xorm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package builder
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func (b *Builder) limitWriteTo(w Writer) error {
+	if strings.TrimSpace(b.dialect) == "" {
+		return errors.New("field `dialect` must be set up when performing LIMIT, try use `Dialect(dbType)` at first")
+	}
+	if !(b.optype == selectType || b.optype == unionType) {
+		return errors.New("LIMIT operation is limited in SELECT and UNION")
+	}
+
+	if b.limitation != nil {
+		limit := b.limitation
+		if limit.offset < 0 || limit.limitN <= 0 {
+			return errors.New("unexpected offset/limitN")
+		}
+
+		selects := b.selects
+		// erase limit condition
+		b.limitation = nil
+		// flush writer, both buffer & args
+		ow := w.(*BytesWriter)
```

Comment:
If Dialect is `SQLITE, MYSQL, POSTGRES`, it's unnecessary to reset the writer.

Reply:
Okay, I'll change it

Reply:
As above

Reply:
Okay, I'll change it